addons/mail/static/src/core/common/navigable_list.xml

```diff
@@ -12,16 +12,19 @@
                 </div>
                 <t t-else="">
                     <div
-                        t-foreach="props.options" t-as="option" t-key="option_index"
+                        t-foreach="sortedOptions" t-as="option" t-key="option_index"
                         class="o-mail-NavigableList-item"
                         t-att-class="option.classList"
                         t-on-mouseenter="() => this.onOptionMouseEnter(option_index)"
                         t-on-click="(ev) => this.selectOption(ev, option_index)"
                     >
+                        <hr class="my-2" t-if="option.group != lastGroup and option_index != 0"/>
                         <a href="#" class="d-flex align-items-center w-100 py-2 px-4" t-att-class="{ 'o-mail-NavigableList-active bg-300': state.activeIndex === option_index }">
-                            <t t-if="props.optionTemplate" t-call="{{ props.optionTemplate }}"/>
+                            <t t-if="option.optionTemplate" t-call="{{ option.optionTemplate }}"/>
+                            <t t-elif="props.optionTemplate" t-call="{{ props.optionTemplate }}"/>
                             <t t-else="" t-esc="option.label"/>
                         </a>
+                        <t t-set="lastGroup" t-value="option.group"/>
```

Comment:
confused about the name and the usage of this. it's the item being looped right? then why "last"?

Reply:
I keep track of the group of the last item and compare it with the current one's group in order to conditionally display the group separator.
If the groups are different I display the separator.